File: benchmarks/web_apis/http_get.js
const http = require('http');
const server = http.createServer((req, res) => { res.end('ok'); });
server.listen(0, '127.0.0.1', () => {
    const port = server.address().port;
    let done = 0;
    const start = Date.now();
    for (let i = 0; i < 100; i++) {
        http.get(`http://127.0.0.1:${port}/`, (res) => {
            let d = '';
            res.on('data', c => d += c);
            res.on('end', () => {
                done++;
                if (done === 100) {
                    console.log((Date.now() - start) * 1000);
                    server.close();
                }
            });
        });
    }
});

File: benchmarks/web_apis/http_get.go
package main

import (
	"fmt"
	"io"
	"net"
	"net/http"
	"time"
)

func main() {
	mux := http.NewServeMux()
	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte("ok"))
	})
	listener, _ := net.Listen("tcp", "127.0.0.1:0")
	srv := &http.Server{Handler: mux}
	go srv.Serve(listener)
	addr := listener.Addr().String()

	start := time.Now()
	for i := 0; i < 100; i++ {
		resp, _ := http.Get("http://" + addr + "/")
		io.ReadAll(resp.Body)
		resp.Body.Close()
	}
	fmt.Println(time.Since(start).Microseconds())
	srv.Close()
}
